packages/protocol/contracts/layer1/verifiers/compose/ComposeVerifier.sol

```diff
@@ -52,24 +62,42 @@ abstract contract ComposeVerifier is IProofVerifier {
     function verifyProof(bytes32 _transitionsHash, bytes calldata _proof) external view {
         SubProof[] memory subProofs = abi.decode(_proof, (SubProof[]));
         uint256 size = subProofs.length;
-        address[] memory verifiers = new address[](size);
+        uint8[] memory verifierIds = new uint8[](size);
 
-        address verifier;
+        uint8 lastVerifierId = 0;
 
         for (uint256 i; i < size; ++i) {
-            require(subProofs[i].verifier != address(0), CV_INVALID_SUB_VERIFIER());
-            require(subProofs[i].verifier > verifier, CV_INVALID_SUB_VERIFIER_ORDER());
+            uint8 verifierId = subProofs[i].verifierId;
+
+            require(verifierId != NONE, CV_INVALID_SUB_VERIFIER());
+            require(verifierId > lastVerifierId, CV_INVALID_SUB_VERIFIER_ORDER());
+
+            address verifier = getVerifierAddress(verifierId);
+            require(verifier != address(0), CV_INVALID_SUB_VERIFIER());
 
-            verifier = subProofs[i].verifier;
             IProofVerifier(verifier).verifyProof(_transitionsHash, subProofs[i].proof);
 
-            verifiers[i] = verifier;
+            verifierIds[i] = verifierId;
+            lastVerifierId = verifierId;
         }
 
-        require(areVerifiersSufficient(verifiers), CV_VERIFIERS_INSUFFICIENT());
+        require(areVerifiersSufficient(verifierIds), CV_VERIFIERS_INSUFFICIENT());
+    }
+
+    /// @notice Returns the verifier address for a given verifier ID
+    /// @param _verifierId The verifier ID to query
+    /// @return The address of the verifier (or address(0) if invalid)
+    function getVerifierAddress(uint8 _verifierId) public view returns (address) {
```

Comment:
I prefer to expose the immutables as public and then we can easily see all the address on etherscan to konow which verifiers we are actually using